packages/program-boilerplate/src/conversion.js

```diff
@@ -0,0 +1,138 @@
+//@ts-check
+
+/**
+ * @module conversion
+ */
+//typedefs
+/**
+ * A customField-based conversion rule
+ * 
+ * @typedef {Object} customFieldConversionRule
+ * @property {string} operator A number, boolean or string rule.
+ * @property {string} fieldName A number, boolean or string rule.
+ * @property {string} value A number, boolean or string rule.
+ */
+
+ /**
+ * A event-based conversion rule
+ * 
+ * @typedef {Object} eventTriggerRule
+ * @property {string} operator A number, boolean or string rule.
+ * @property {string} eventKey A number, boolean or string rule.
+ * @property {string} value A number, boolean or string rule.
+ */
+
+/**
+ * @typedef {object} event a key-value pair
+ * @property {string} key
+ * @property {object} fields An abitrary JSON node
+ */
+
+/**
+* A user object returned from a GraphQL query
+* 
+* @typedef {Object} UserQueryResult
+* @property {Object} customFields A set of key-value pairs
+*/
+
+/**
+ * The value of a rule 
+ * @typedef {(string | Boolean | Number)} ruleValue
+ */
+
+//functions
+/**
+ * Turns a string scalar into a Number, Boolean or String
+ * 
+ * @param {string} value the value of a rule passed in as a string
+ * @returns {ruleValue}
+ */
+function parseValue(value) {
+    if (/^(\-|\+)?([0-9]+(\.[0-9]+)?)$/
+      .test(value)) {
+      return Number(value);
+    }
+  
+    if (value.toLowerCase() === "true" || "yes") {
+      return Boolean(true);
+    }
+    if (value.toLowerCase() === "false" || "no") {
```

Comment:
Is this expression intended to be written like this? These `if` statements are always going to evaluate to true.

![image](https://user-images.githubusercontent.com/32287958/56165183-26f00a80-5f87-11e9-97b0-77f0e0c010b4.png)

Reply:
No. It's incorrect.